infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/server/secure/jwtproxy/PassThroughProxyProvisioner.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.workspace.infrastructure.kubernetes.server.secure.jwtproxy;
+
+import static java.util.Collections.singletonList;
+
+import com.google.inject.assistedinject.Assisted;
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.security.NoSuchAlgorithmException;
+import javax.inject.Inject;
+import javax.inject.Named;
+import org.eclipse.che.api.core.model.workspace.config.ServerConfig;
+import org.eclipse.che.api.core.model.workspace.runtime.RuntimeIdentity;
+import org.eclipse.che.api.workspace.server.spi.InternalInfrastructureException;
+import org.eclipse.che.workspace.infrastructure.kubernetes.server.external.ExternalServiceExposureStrategy;
+import org.eclipse.che.workspace.infrastructure.kubernetes.server.secure.jwtproxy.factory.JwtProxyConfigBuilderFactory;
+
+/**
+ * A Jwt Proxy provisioner that configures JWT proxy to just pass through all the traffic without
+ * any authentication. Should be used in the single user mode only for "securing" the secure
+ * servers.
+ */
+public class PassThroughProxyProvisioner extends AbstractJwtProxyProvisioner {
+
+  @Inject
+  public PassThroughProxyProvisioner(
+      JwtProxyConfigBuilderFactory jwtProxyConfigBuilderFactory,
+      ExternalServiceExposureStrategy externalServiceExposureStrategy,
+      CookiePathStrategy cookiePathStrategy,
+      @Named("che.server.secure_exposer.jwtproxy.image") String jwtImage,
+      @Named("che.server.secure_exposer.jwtproxy.memory_limit") String memoryLimitBytes,
+      @Named("che.workspace.sidecar.image_pull_policy") String imagePullPolicy,
+      @Assisted RuntimeIdentity identity)
+      throws InternalInfrastructureException {
+    super(
+        constructSignatureKeyPair(),
+        jwtProxyConfigBuilderFactory,
+        externalServiceExposureStrategy,
+        cookiePathStrategy,
+        jwtImage,
+        memoryLimitBytes,
+        imagePullPolicy,
+        identity.getWorkspaceId(),
+        false);
+  }
+
+  /**
+   * Constructs a key pair to satisfy JWT proxy which needs a key pair in its configuration. In case
+   * of pass-through proxy, this key pair is unused so we just generate a random one.
```

Comment:
how hard would it be to teach JWT proxy to act as a pass-through proxy without dummy key magic?

Reply:
There is another kind of verifier "key server" that this key boils down to being required for - it is called "keyregistry" and requires just a URL, which we could provide with just some dummy value I believe. But we would have to do quite a big changes to `JwtProxyConfigBuilder` to support different kinds of key servers than just the currently used "preshared" (which is the thing requiring the preshared ssh key).

So IMHO generating the dummy keypair is cheaper and safer than rewriting the config builder.